ES|QL: Add note in Known Issues for #130644 (#130996)
elastic/elasticsearch: docs/release-notes/known-issues.md

Known issues are significant defects or limitations that may impact your implementation. These issues are actively being worked on and will be addressed in a future release. Review the Elasticsearch known issues to help you make informed decisions, such as upgrading to a new version.

## 9.0.3 [elasticsearch-9.0.3-known-issues]
* A bug in the merge scheduler in Elasticsearch 9.0.3 may prevent shards from closing when there isn’t enough disk space to complete a merge. As a result, operations such as closing or relocating an index may hang until sufficient disk space becomes available.
To mitigate this issue, the disk space checker is disabled by default in 9.0.3 by setting `indices.merge.disk.check_interval` to `0` seconds. Manually enabling this setting is not recommended.

>   This issue is planned to be fixed in future patch release [#129613](https://github.com/elastic/elasticsearch/pull/129613)

Reviewer: @luigidellaquila I think the _best_ practice is to add the known issue to all affected versions

>   This issue is planned to be fixed in future patch release [#129613](https://github.com/elastic/elasticsearch/pull/129613)

Author: > I think the best practice is to add the known issue to all affected versions

OK, I'm adding the same note in the 9.0.0 section.

> I wonder if we added a known issues section (or link) in the ESQL limitations page that would be helpful too?

I think it makes sense. Probably, in terms of maintenance effort, a link to the known issues is easier to manage. My concern is that we could end up with misalignments if we duplicate information.

I'm adding a new section, please have a look if you have a chance.

>   This issue is planned to be fixed in future patch release [#129613](https://github.com/elastic/elasticsearch/pull/129613)

Reviewer: LGTM

> * A bug in ES|QL STATS command may lead to wrong results. The bug only happens in a very specific case, ie. with `STATS ... BY keyword1, keyword2`:
> exactly two grouping fields, both keywords, where the first field has high cardinality (more than 65k distinct values).

Reviewer: ```suggestion
* A bug in the ES|QL STATS command may yield incorrect results. The bug only happens in very specific cases that follow this pattern: `STATS ... BY keyword1, keyword2`, i.e. the command must have exactly two grouping fields, both keywords, where the first field has high cardinality (more than 65k distinct values).
```

>   The issue is described in detail in [this issue](https://github.com/elastic/elasticsearch/issues/130644).

Reviewer: ```suggestion
  The bug is described in detail in [this issue](https://github.com/elastic/elasticsearch/issues/130644).
```

>   The problem was introduced in 8.16.0 and [fixed](https://github.com/elastic/elasticsearch/pull/130705) in 8.17.9, 8.18.7, 9.0.4,

Reviewer: ```suggestion
  The problem was introduced in 8.16.0 and [fixed](https://github.com/elastic/elasticsearch/pull/130705) in 8.17.9, 8.18.7, 9.0.4.
```

>   Possible work-arounds are:

Reviewer: ```suggestion
  Possible workarounds include:
```

  * switching the order of the grouping keys (eg. `STATS ... BY keyword2, keyword1`, if the `keyword2` has a lower cardinality)

>   * reducing the grouping key cardinality, filtering out values before STATS.

Reviewer: ```suggestion
  * reducing the grouping key cardinality, by filtering out values before STATS
```
